src/main/java/frc/robot/subsystems/Revolver.java

```diff
@@ -56,16 +41,27 @@ public void setAll(double drumSpeed, double feedWheelSpeed) {
     setFeedWheel(feedWheelSpeed);
   }
 
+  public void resetReps(){
+    m_drumReps = 0;
+  }
+
   /**
    * Sets the speed of the drum
    * @param speed the speed to run the drum at as a percentage from -1 to 1
    */
   public void setDrum(double speed) {
     m_drumMotor.set(speed);
-    if (m_pdp.getCurrent(RobotMap.PDP.DRUM_PORT) > Constants.REVOLVER_DRUM_CURRENT_THRESHOLD) {
-      m_repsAboveDrum++;
-    } else {
-      resetDrumReps();
+
+    if((speed==0) || (speed!=0 && m_drumMotor.getEncoder().getVelocity()!=0)){
+      resetReps();
+    }
+
+    if (speed > 0 && m_drumMotor.getEncoder().getVelocity()==0.0){
+      m_drumReps++;
+    }
+    if(m_drumReps > Constants.REVOLVER_REPS_THRESHOLD){
+      new UnjamRevolver(this).schedule(false);
+      resetReps();
     }
```

Comment:
This is a lot better, but we can still simplify this logic by using an `else`. Consider the following code:

```
if (speed > 0.0 && m_drumMotor.getEncoder.getVelocity == 0.0) {
  m_drum_reps++;
} else {
  resetReps();
}

if (m_drumReps > Constants.REVOLVER_REPS_THRESHOLD) {
  resetReps();
  new UnjamRevolver(this).schedule(false);
}
```

I would also like to move this back into `periodic`. @Jacob4649 Are you okay with this?

Reply:
Yeah that looks fine to me

Reply:
Why bother with the `Math.max` here? Your code is equivalent either way.